packages/mobile/src/navigator/NavigationService.ts

```diff
@@ -115,14 +115,16 @@ async function ensurePincode(): Promise<boolean> {
       return false
     }
   }
+
   return true
 }
 
 export const navigateProtected: SafeNavigate = (...args) => {
   ensurePincode()
     .then((ensured) => {
       if (ensured) {
-        navigate(...args)
+        const [routeName, params] = args
+        navigationRef.current?.dispatch(StackActions.replace({ routeName, params }))
```

Comment:
This looks like it should take two arguments (https://reactnavigation.org/docs/stack-actions/#replace) - how does it work?

Reply:
I hesitate to say it, but maybe their documentation is wrong?

When you click into the method definition, here's what you see:
```
export declare function replace({ routeName, params, key, newKey, action, }: {
    routeName: string;
    params?: object;
    key?: string;
    newKey?: string;
    action?: never;
}): StackActionType;
```

Reply:
I just realized we have a `replace` function defined (the same way) in the same file, so I am now using that instead

Reply:
ahhhh. we are importing from `@react-navigation/compat`instead of `@react-navigation/native`

Reply:
oh interesting, didnt notice that!